kratos/tests/test_linear_constraints.py

```diff
@@ -18,98 +20,76 @@ def setUp(self):
-            cl = KratosMultiphysics.StructuralMechanicsApplication.LinearElasticPlaneStrain2DLaw(
-            )
+            cl = SMA.LinearElasticPlaneStrain2DLaw()
         else:
-            cl = KratosMultiphysics.StructuralMechanicsApplication.LinearElastic3DLaw(
-            )
-        self.mp.GetProperties()[1].SetValue(KratosMultiphysics.CONSTITUTIVE_LAW, cl)
+            cl = SMA.LinearElastic3DLaw()
+        self.mp.GetProperties()[1].SetValue(KM.CONSTITUTIVE_LAW, cl)
 
     def _apply_BCs(self):
         bcs = self.mp.GetSubModelPart("FixedEdgeNodes")
-        KratosMultiphysics.VariableUtils().SetVariable(
-            KratosMultiphysics.DISPLACEMENT_X, 0.0, bcs.Nodes)
-        KratosMultiphysics.VariableUtils().SetVariable(
-            KratosMultiphysics.DISPLACEMENT_Y, 0.0, bcs.Nodes)
+        KM.VariableUtils().SetVariable(KM.DISPLACEMENT_X, 0.0, bcs.Nodes)
+        KM.VariableUtils().SetVariable(KM.DISPLACEMENT_Y, 0.0, bcs.Nodes)
 
-        KratosMultiphysics.VariableUtils().ApplyFixity(
-            KratosMultiphysics.DISPLACEMENT_X, True, bcs.Nodes)
-        KratosMultiphysics.VariableUtils().ApplyFixity(
-            KratosMultiphysics.DISPLACEMENT_Y, True, bcs.Nodes)
+        KM.VariableUtils().ApplyFixity(KM.DISPLACEMENT_X, True, bcs.Nodes)
+        KM.VariableUtils().ApplyFixity(KM.DISPLACEMENT_Y, True, bcs.Nodes)
 
         bcmn = self.mp.GetSubModelPart("MovingNodes")
-        KratosMultiphysics.VariableUtils().SetVariable(
-            KratosMultiphysics.DISPLACEMENT_X, 0.01, bcmn.Nodes)
-        KratosMultiphysics.VariableUtils().SetVariable(
-            KratosMultiphysics.DISPLACEMENT_Y, 0.0, bcmn.Nodes)
-        KratosMultiphysics.VariableUtils().ApplyFixity(
-            KratosMultiphysics.DISPLACEMENT_X, True, bcmn.Nodes)
-        KratosMultiphysics.VariableUtils().ApplyFixity(
-            KratosMultiphysics.DISPLACEMENT_Y, True, bcmn.Nodes)
+        KM.VariableUtils().SetVariable(KM.DISPLACEMENT_X, 0.01, bcmn.Nodes)
+        KM.VariableUtils().SetVariable(KM.DISPLACEMENT_Y, 0.0, bcmn.Nodes)
+        KM.VariableUtils().ApplyFixity(KM.DISPLACEMENT_X, True, bcmn.Nodes)
+        KM.VariableUtils().ApplyFixity(KM.DISPLACEMENT_Y, True, bcmn.Nodes)
 
-    def _setup_solver(self):
+    def _setup_solver(self, solving_with = "Block", linear_solver = "AMGCL"):
 
         #define a minimal newton raphson solver
-        self.linear_solver = KratosMultiphysics.SkylineLUFactorizationSolver()
-        self.builder_and_solver = KratosMultiphysics.ResidualBasedBlockBuilderAndSolver(
-            self.linear_solver)
-        self.scheme = KratosMultiphysics.ResidualBasedBossakDisplacementScheme(
-            -0.01)
-        self.convergence_criterion = KratosMultiphysics.ResidualCriteria(
-            1e-10, 1e-12)
+        if linear_solver == "AMGCL":
+            self.linear_solver = KM.AMGCLSolver()
+        else:
+            self.linear_solver = KM.SkylineLUFactorizationSolver()
+        if solving_with == "Block":
```

Comment:
this is not necessary

Reply:
It will not be at #5939 ;) (the point is that with the doble LM it is possible to use iterative solvers )